### PR TITLE
Upgrade to elasticsearch v8.7.0

### PR DESCRIPTION
**Pull request by MarvinZeising**
I'm proposing an upgrade to elasticsearch v8.7.0, as requested in #868.

The tests didn't finish locally (even before my changes), so I'm interested to see what the CI action says.

### CODE REVIEW

**Review by orlangure — COMMENTED** [review]
Thank you for working on this!

The tests indeed fail now because the container never becomes healthy. I guess something is changed in the API so we can't use the same method to test container readiness. If you still want to debug this, I suggest starting with `WithLogWriter` and `WithDebugMode` options to look into what happens inside the container, and where does the flow get stuck. 

If there was a change in the API, we'll have to support multiple health checks depending on preset version.

**Comment by orlangure** [comment]
Hey @MarvinZeising,
Turns out elastic tests were garbage and only tested a single version (7), as you just pointed out in your PR. 

It means that only one major version was supported, and versions 5 and 6 were not. As I don't use Elastic in my current projects, I'm not even sure how important is this support. I think that maybe version 6 is somewhat important to support, but 5 can be dropped...

Anyway, if you decide to work on supporting 7 and 8 side by side (in code and in tests), maybe you could look into supporting version 5 and 6, or only 6, at the same time. For now, I modified the tests to only run on version 7, and removed 5 and 6 from the list of supported versions in readme.

File: preset/elastic/preset.go [hunk 1]
```diff
@@ -26,11 +26,11 @@ func init() {
 	registry.Register("elastic", func() gnomock.Preset { return &P{} })
 }
 
-// Preset creates a new Gmomock Elasticsearch preset. This preset includes an
+// Preset creates a new Gnomock Elasticsearch preset. This preset includes an
 // Elasticsearch specific healthcheck function and default Elasticsearch image
 // and port.
 //
-// By default, version 7.9.3 is used.
+// By default, version 8.7.0 is used.
```
[the remark this answers is not in the document]
Comment (MarvinZeising) [reply preset/elastic/preset.go:33]: Totally overlooked that, thanks!

**Comment by MarvinZeising** [comment]
Thanks for your review @orlangure.

Versions 5 and 6 of elasticsearch are already at end of life. If they're not supported officially and no one is using them (because they didn't work with gnomork before), I don't think we should add this.
Version 7.9.3 is also at end of life, so I updated it to the last v7 which is 7.17.9. That should be backwards-compatible with the previous minor versions.

The container wasn't getting healthy because the endpoint is secure. Disabling TLS fixed. Since in my company, we use that setting in v7 as well (because it's a sensible default setting) I didn't even add a version check.

I tried to combine the tests for the different versions like you did. But that proved to be difficult since there's no common interface for the clients. So I figured the cleanest approach would just be to separate these tests and use the respective client library.
Is that okay for you?

**Comment by orlangure** [comment]
I don't mind separating the tests, if this is easier.

But something confuses me. We now use client v8 in the preset regardless of versions (7 or 8). Does this client know how to connect and work with both Elastic 7 and 8? Because a server test for example fails because it sends v7.9.2 as the version, and Elastic client v8 can't connect to it. I would expect that existing tests continue to work if they use v7. So something wasn't backwards compatible between 7.9 and 7.17. What do you think we should do with this issue?

**Comment by MarvinZeising** [comment]
Thanks for your comment, @orlangure!

I missed two places to update the default elasticsearch version in. I fixed that and now the tests succeed locally.

My idea was that `preset_test.go` tests the default version (currently 8.7.0) and `preset_<version>_test.go` tests a specific supported older version. That's why the first one uses the v8 client, while the other test uses the v7 client.

I also tried the server test with 7.17.9 (by specifying that in `elastic.json` > `preset.version`) and the v8 client is indeed backwards-compatible with that version, just not with 7.9 as that seems to be too old.

**Review by orlangure — APPROVED** [review]
Amazing, thank you for the contribution!

**Pull request merged**